felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1122,87 +1131,37 @@ func (m *bpfEndpointManager) onInterfaceUpdate(update *ifaceStateUpdate) {
 		return
 	}
 
+	// Should be safe without the lock since there shouldn't be any active background threads
```

Comment:
why did you move this down here?

Reply:
I take the lock in `cleanupOldAttach` to get the policyIdx info from the state. Hence moved the lock in the calling function below the block where we call `cleanupOldAttach`